lib/jekyll/tags/include.rb

```diff
@@ -213,8 +213,15 @@ def page_path(context)
         if context.registers[:page].nil?
           context.registers[:site].source
         else
-          current_doc_dir = File.dirname(context.registers[:page]["path"])
-          context.registers[:site].in_source_dir current_doc_dir
+          site = context.registers[:site]
+          page_payload  = context.registers[:page]
+          resource_path = \
+            if page_payload.is_a?(Jekyll::Drops::DocumentDrop)
```

Comment:
What do you think about `!page_payload[“collection”].nil?` instead of checking the class name? That would be more future proof :)

Reply:
Awesome! I like that suggestion..